pkg/controller/storagecluster/util.go

```diff
@@ -0,0 +1,108 @@
+package storagecluster
+
+import (
+	"encoding/binary"
+	"fmt"
+	"hash/fnv"
+
+	corev1alpha1 "github.com/libopenstorage/operator/pkg/apis/core/v1alpha1"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
+	schedapi "k8s.io/kubernetes/pkg/scheduler/api"
+	hashutil "k8s.io/kubernetes/pkg/util/hash"
+)
+
+func addOrUpdateStoragePodTolerations(pod *v1.Pod) {
+	// StorageCluster pods shouldn't be deleted by NodeController in case of node problems.
+	// Add infinite toleration for taint notReady:NoExecute here to survive taint-based
+	// eviction enforced by NodeController when node turns not ready.
+	v1helper.AddOrUpdateTolerationInPod(pod, &v1.Toleration{
```

Comment:
We should also set ourselves as critical pods: https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/

Reply:
Adding a TODO for this. Will explore more with system-cluster-critical and system-node-critical. The annotation is getting removed from k8s 1.14.